pipeline/process_block.go

```diff
@@ -179,14 +214,26 @@ func (p *Pipeline) handlerStepNew(ctx context.Context, block *bstream.Block, clo
 	}
 
 	if reqDetails.ShouldReturnProgressMessages() {
-		if err = p.returnModuleProgressOutputs(clock); err != nil {
-			return fmt.Errorf("failed to return modules progress %w", err)
+		if reqDetails.IsSubRequest {
+			if err = p.returnInternalModuleProgressOutputs(clock); err != nil {
+				return fmt.Errorf("failed to return modules progress %w", err)
+			}
+		} else {
+			if err = p.returnRPCModuleProgressOutputs(clock); err != nil {
+				return fmt.Errorf("failed to return modules progress %w", err)
+			}
 		}
 	}
 
 	if p.gate.shouldSendOutputs() {
 		logger.Debug("will return module outputs")
-		if err = returnModuleDataOutputs(clock, bstream.StepNew, cursor, p.moduleOutputs, p.respFunc); err != nil {
+		for _, hook := range p.preFirstBlockDataHooks {
+			if err := hook(ctx, clock); err != nil {
+				return fmt.Errorf("failed to run pre-first-block-data-hook: %w", err)
```

Comment:
```suggestion
				return fmt.Errorf("failed to run pre first block data hook: %w", err)
```

I think we are safe to use the spaced version.